src/horizontal-pod-autoscaler.ts

```diff
@@ -0,0 +1,866 @@
+import { ApiObject, Duration, Lazy } from 'cdk8s';
+import { Construct } from 'constructs';
+import * as kplus from '../src';
+import { Resource, ResourceProps } from './base';
+import { Container } from './container';
+import * as k8s from './imports/k8s';
+import { Workload } from './workload';
+
+/**
+ * Properties for HorizontalPodAutoscaler
+ */
+export interface HorizontalPodAutoscalerProps extends ResourceProps {
+  /**
+   * The workload to scale up or down.
+   *
+   * Scalable workload types:
+   * * Deployment
+   * * ReplicaSet
+   * * ReplicationController
+   * * StatefulSet
+   *
+   * Cannot be used with workloads that do not scale, such as DaemonSet.
+   */
+  readonly target: Workload;
+  /**
+   * The maximum number of replicas that can be scaled up to.
+   */
+  readonly maxReplicas: number;
+  /**
+   * The minimum number of replicas that can be scaled down to.
+   *
+   * Can be set to 0 if the alpha feature gate `HPAScaleToZero` is enabled and
+   * at least one Object or External metric is configured.
+   *
+   * @default 1
+   */
+  readonly minReplicas?: number;
+  /**
+   * The metric conditions that trigger a scale up or scale down.
+   *
+   * @default
+   * No metrics are configured. Only scales on the requests or limits defined on the target.
```

Comment:
Our convention is to use `default -` when we describe a behavior.

```suggestion
   * @default - No metrics are configured. Only scales on the resource requests or limits defined on the target.
```

Also, what is the operator of the conditions? OR? AND?